### ISSUE TITLE
Can't backward when training datasets'identities are more than 10000.

### ISSUE THREAD

**Issue by maryhh** [naimple/regular-face-cvpr2019#1]
Hello,
i use this method to train,but i found that it didn't backward and the training data has 60000 ids.
Casia_webface is ok.Do you know the reason?
here is my code in mxnet:
def forward(self,is_train,req,in_data,out_data,aux):
    weight = in_data[0]
    cos = mx.nd.dot(weight,weight.T)
    cos[np.arange(self.numclasses),np.arange(self.numclasses)] = -100.0
    index = mx.nd.argmax(cos,axis=0)
    exclusive_loss = cos[np.aramge(self.numclasses),index]
    exclusive_loss = mx.nd.summ(exclusive_loss)/self.numclasses
    self.assign(out_data[0],req[0],exclusive_loss)
def backward(self,req,out_grad,in_data,out_data,in_grad,aux):
    diff = aux[0]
    self.assign(in_grad[0],req[0],diff)

**Comment by williechai** [comment]
> Hello,
> i use this method to train,but i found that it didn't backward and the training data has 60000 ids.
> Casia_webface is ok.Do you know the reason?
> here is my code in mxnet:
> def forward(self,is_train,req,in_data,out_data,aux):
> weight = in_data[0]
> cos = mx.nd.dot(weight,weight.T)
> cos[np.arange(self.numclasses),np.arange(self.numclasses)] = -100.0
> index = mx.nd.argmax(cos,axis=0)
> exclusive_loss = cos[np.aramge(self.numclasses),index]
> exclusive_loss = mx.nd.summ(exclusive_loss)/self.numclasses
> self.assign(out_data[0],req[0],exclusive_loss)
> def backward(self,req,out_grad,in_data,out_data,in_grad,aux):
> diff = aux[0]
> self.assign(in_grad[0],req[0],diff)

this is big matrix multiplication in this regular loss (weight * weight.t),  neigher vggface2 nor webface is a big-identity dataset. In my test, when i do this matrix mulplication on (10000, 256) and its transpose, my 1080 has been taken away nearly half graphics memory, i won't believe i can train MS_1M_Celeb or MegaFace with RegularFace loss.
